elegy/nets/resnet.py

```diff
@@ -141,22 +141,31 @@ def __init__(
 
         if weights is not None:
             if weights.endswith(".pkl"):
-                parameters = pickle.load(open(weights, "rb"))
+                collections = pickle.load(open(weights, "rb"))
             elif weights == "imagenet":
                 clsname = self.__class__.__name__
                 urldict = PRETRAINED_URLS.get(clsname, None)
                 if urldict is None:
                     raise ValueError(f"No pretrained weights for {clsname} available")
                 fname = utils.download_file(urldict["url"], sha256=urldict["sha256"])
-                parameters = pickle.load(open(fname, "rb"))
+                collections = pickle.load(open(fname, "rb"))
             else:
                 raise ValueError("Unknown weights value: ", weights)
 
+            if isinstance(collections, tuple):
+                parameters, collections = collections
+            elif "parameters" in collections:
+                parameters = collections.pop("parameters")
+            else:
```

Comment:
@alexander-g added the cases above so its compatible with the previous output of `get_default_parameters` (a dict with a "parameters" key) and the new version (a tuple (params, collections)). Feel free to clean this up if you update the weights.